### DISCUSSION TITLE
next image invalid src prop on localhost

### DISCUSSION THREAD

**Discussion by ivanvinicius** [vercel/next.js#56390] [Help]
### Summary

I have already configured CORS in my api

I facing a problem trying to access some images on my localhost api (port 3333), you can check my next.config file down below:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  images: {
    domains: ['localhost'],
  },
}

module.exports = nextConfig
```

My packege.json
```js
  "dependencies": {
    "@tanstack/react-query": "^4.35.3",
    "axios": "^1.5.0",
    "lucide-react": "^0.279.0",
    "next": "latest", // which is 13.5 with pages router
    "react": "latest",
    "react-dom": "latest"
  },
```

The erro message is:
```cmd
Error: Invalid src prop (localhost:3333/uploads/nationalities/11d03fcdb97bcf6d8981-usa.png) on `next/image`, hostname "" is not configured under images in your `next.config.js`
See more info: https://nextjs.org/docs/messages/next-image-unconfigured-host
```

I have tried remotePatterns too

Do you have any idea?

**Comment by skve** [comment]
The missing hostname in the error is interesting, but the issue is probably the lack of the port.

If you use:

```ts
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  images: {
    domains: ['localhost:3333'],
  },
}

module.exports = nextConfig
```

It will probably work. I wouldn't recommend it in production though, you're better off pointing towards your server's hostname rather than localhost.

**Reply by ivanvinicius** [reply]
In production I'm use aws s3 (that works correctly), localhost is just for development proposal.

I have tried some things like you metioned, but no success. I don't know why but hostname is missing in every attempt

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  images: {
    // domains: ['localhost:3333'],
    remotePatterns: [
      {
        protocol: 'http',
        hostname: 'localhost',
        port: '3333',
        pathname: '**',
      },
    ],
  },
}

module.exports = nextConfig


```

**Comment by forhadkhan** [comment]
I faced the same problem, and in my case, this configuration worked:

```next.config.mjs
/** @type {import('next').NextConfig} */

const nextConfig = {
  images: {
    remotePatterns: [
      {
        protocol: "http",
        hostname: "localhost",
        port: "3000",
        pathname: "/**",
      },
    ],
  },
};

export default nextConfig;
```

Make sure to use a slash in the `pathname`.

---------

The following configuration also works:

```.js
const nextConfig = {
  images: {
    domains: ['localhost'],
  },
};

export default nextConfig;
```
However, this approach is **deprecated**, so it’s best to use the updated method.

(You might not need this answer anymore, but hopefully, it helps someone else.)
